cabal-dev-scripts/src/GenUtils.hs

```diff
@@ -162,6 +168,12 @@ toConstructorName t = t
     special :: Text -> Text
     special "0BSD"          = "NullBSD"
     special "389_exception" = "DS389_exception"
+    -- General way to handle licences starting with a digit.
+    -- We cannot remove the two patterns above as it would be
+    -- a needless breaking change.
+    special u
+      | Just (c, _) <- T.uncons u
+      , C.isDigit c = "NN_" <> u
```

Comment:
This bit just prepends "NN_" to strings starting with a digit.

I wonder if it makes better sense than handling each string in an ad-hoc basis (and if it makes better sense, whether we should  also let "0BSD" and "389_exception" become "NN_0BSD" and "NN_389_exception").

Reply:
Yeah let's adopt the NN_ convention (I just hope nobody registers the NNN license…)